test/infrastructure/docker/exp/internal/controllers/dockermachinepool_controller.go

```diff
@@ -109,6 +112,18 @@ func (r *DockerMachinePoolReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		return ctrl.Result{}, err
 	}
 
+	dockerMachinePool.Status.InfrastructureMachineSelector = metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			clusterv1.ClusterNameLabel:            cluster.Name,
+			infraexpv1.DockerMachinePoolNameLabel: dockerMachinePool.Name,
+		},
+	}
+	dockerMachinePool.Status.InfrastructureMachineKind = "DockerMachine"
+	// Patch now so that the status and selectors are available.
+	if err := patchHelper.Patch(ctx, dockerMachinePool); err != nil {
```

Comment:
Why is it important to patch this immediately instead of at the end of the reconcile in defer?

Reply:
The MP controller is unable to tell that we support MPMs until this gets set by the controller, so the earlier we do it the less downtime there will be.

Reply:
Frankly speaking I don't think this is relevant in term of downtime, the time from this patch call and the defer patch call in the main is usually neglectable (also there is no downtime, it is just the system converging to the desired state)

But if we have to do this let's wrap it into a 
`if dockerMachinePool.Status.InfrastructureMachineKind != "DockerMachine"`